### DISCUSSION TITLE
Rule Proposal: HTML Tag Requires Attribute

### DISCUSSION THREAD

**Discussion by enesyalcin** [angular-eslint/angular-eslint#1800] [New Rule Proposals]
Hi!

We have to add ids to Buttons and iFrames wherever they occur so the testing team can access them easier. Therefore, it would be great if we could have the Rule like `@html-eslint` has for requiring certain Attributes for HTML Tags.

Proposed Rule name: `html-tag-require-attribute`

Example as how `@html-eslint` did it. There you can configure which HTML Tags should have which Attributes.

[@html-eslint/require-attrs](https://html-eslint.org/docs/rules/require-attrs)

```
  rules: {
    "@html-eslint/require-attrs": [
      "error",
      {
        tag: "img",
        attr: "alt", // Enforce to use img with alt attribute.
      },
      {
        tag: "svg",
        attr: "viewBox" // Enforce to use svg and viewBox attributes with "0 0 100 100" value.
        value: "0 0 100 100"
      },
    ],
  },
```

I would be open to contribute with a little bit of Help into the right direction.

**Comment by undsoft** [comment]
This may help: https://github.com/undsoft/eslint-plugin-angular-test-ids
